src/cascade/model/dismod_groups.py

```diff
@@ -0,0 +1,60 @@
+from collections import UserDict
+from os import linesep
+
+
+class DismodGroups(UserDict):
+    """Dismod-AT documentation discusses five kinds of groups of
+    model variables (https://bradbell.github.io/dismod_at/doc/model_variables.htm).
+    This class represents that grouping as a set of dictionaries, where the
+    values can either be SmoothGrids or RandomFields or whatever is classified
+    according to groups of model variables.
+
+     * Rate key is rate as a string (iota, rho, chi, omega, pini)
+     * Random effect key is tuple (rate, location_id), where None is all.
+     * Alpha key is (covariate, rate), both as strings.
+     * Beta key is (covariate, integrand), both as strings.
```

Comment:
This makes me a little nervous. Especially since the Beta key and the Gamma key are the same. But I see that you're storing the groups separately...

My impulse is to ask for something like:
```
my_dismod_groups[Gamma("covariate", "integrand")] = whatever
```
But that adds complexity over what you have and I don't know it really gets us anything. Maybe this:

```
my_dismod_groups.gamma[("covariate", "integrand")] = somethig
```

Reply:
You could still get a clean iter over the different groups if that's why you're choosing the nested dict like structure in the first place. Would be a short implementation.

Reply:
This class is pretty obscure to stare at. It ends up giving `my_dismod_groups.gamma[("covariate", "integrand")] = something`. I'd like to add checking for the keys, so that each of rate, re, alpha, beta, gamma is a dictionary that checks its keys. The pattern in model_reader ends up OK as a nested for loop. I'm not complaining about that. That's internal code, written only a few times, and can be harder to write without penalty. I want this good for the user.

I was trying to leave open making this an object with members, so groups.rate.iota and groups.alpha.traffic.iota. There's a little challenge in using numbers for locations, but we could prefix an underscore.

Reply:
Oh, it does have the getattr implementation. That's fine then. Sorry I overlooked that.